chroma-agent/chroma_agent/lib/node_admin.py

```diff
@@ -2,35 +2,39 @@
 # Use of this source code is governed by a MIT-style
 # license that can be found in the LICENSE file.
 
-
 from jinja2 import Environment, PackageLoader
-env = Environment(loader=PackageLoader('chroma_agent', 'templates'))
-
-from iml_common.lib.util import platform_info
 from chroma_agent.lib.shell import AgentShell
 
+env = Environment(loader=PackageLoader('chroma_agent', 'templates'))
+
 
 def write_ifcfg(device, mac_address, ipv4_address, ipv4_netmask):
     ifcfg_tmpl = env.get_template('ifcfg-nic')
     ifcfg_path = "/etc/sysconfig/network-scripts/ifcfg-%s" % device
     with open(ifcfg_path, "w") as f:
-        f.write(ifcfg_tmpl.render(device = device, mac_address = mac_address,
-                                  ipv4_address = ipv4_address,
-                                  ipv4_netmask = ipv4_netmask))
+        f.write(ifcfg_tmpl.render(device=device,
+                                  mac_address=mac_address,
+                                  ipv4_address=ipv4_address,
+                                  ipv4_netmask=ipv4_netmask))
 
     return ifcfg_path
 
 
 def unmanage_network(device, mac_address):
-    """Rewrite the network configuration file to set NM_CONTROLLED="no"
-    TODO: This is destructive and overwrites the file loosing all settings.
-    This needs to be fixed up.
+    """
+    Rewrite the network configuration file to set NM_CONTROLLED="no"
+
+    TODO: This is destructive and overwrites the file clearing
+    previously configured settings, needs fixing.
     """
     ifcfg_path = write_ifcfg(device, mac_address, None, None)
 
-    if platform_info.distro_version >= 7.0:
+    if ifcfg_path:
         try:
             AgentShell.try_run(['nmcli', 'con', 'load', ifcfg_path])
+        except OSError:
+            # Network Manager not installed
+            pass
         except AgentShell.CommandExecutionError as cee:
-            if cee.result.rc not in [127, 2, 8]:            # network manager may be uninstalled (127) stopped (8)
+            if cee.result.rc != 8:  # network manager stopped
```

Comment:
Do we know what the removed 2 and 127 were covering?

Reply:
yes, 2 refers to bad syntax which shouldn't be encountered, 127 would refer to package not being installed in which case it wouldn't reach that line of code